src/eth_provider/provider.rs

```diff
@@ -171,20 +173,25 @@ pub trait EthereumProvider {
 /// Uses access to a database for certain data, while
 /// the rest is fetched from the Starknet Provider.
 #[derive(Debug, Clone)]
-pub struct EthDataProvider<SP: starknet::providers::Provider> {
+pub struct EthDataProvider<SP: starknet::providers::Provider + Send + Sync> {
     database: Database,
     starknet_provider: SP,
     chain_id: u64,
+    pub mempool: Option<KakarotPool<Self>>,
```

Comment:
Not exactly sure, the problem here is that to be able to access the mempool in the `send_raw_transaction`, I would say that the mempool must be in the provider. And we have a kind of circle dep, because the mempool which is inside the provider depends on the provider itself (because of the access to the db accounts): `KakarotPool<Self>` so I put an Option for the moment for the draft to be able to do:
- Initialization of the provider with None mempool
- Initialization of the mempool using the provider
- Addition of the mempool in the provider

Reply:
ah this is annoying... I think we need to try to remove this cercle dep, it's not great to have this

Reply:
fully aligned there; but in the validation of the mempool tx we must have access to the account to have its balance and check it. And unfortunately the only place where we do that is the provider...

The best strategy would surely be to put the mempool in the account manager service but we would have to find a method so that the send raw transaction has access to it too.

Reply:
true.... let's keep it like this for now and make an issue to refactor this, it's not an easy problem